marathon-template

```diff
@@ -28,23 +27,39 @@ def fetch_tasks(marathon_uri, app_id):
   tasks = filter(lambda x: x is not None, map(lambda task: task_info(task["host"], task["ports"]), app_tasks))
   return sorted(tasks, key=lambda task: task.name())
 
-def tasks_for(marathon_url, apps):
-  response = {}
-  for app in apps:
+def fetch_apps(marathon_uri):
+  apps = http(marathon_uri+"/v2/apps?embed=apps.tasks")
+  app_settings = apps["apps"]
+  def app_info(app_config):
     try:
-      response[app] = fetch_tasks(marathon_url, app)
+      return MarathonAppInfo(str(app_config["id"].split("/")[1]), app_config["labels"], app_config["tasks"])
     except:
-      print "Failed fetching task info for %s" % (app)
-      pass
-  return response
+      return None
+  return filter(lambda x: x is not None, map(lambda appConfig: app_info(appConfig), app_settings))
+
+def get_apps_to_monitor(app_configs, app_names_to_monitor):
+  return filter(lambda app_config: app_config.should_monitor(app_names_to_monitor), app_configs)
+
+def get_tasks(app_configs):
+  return {app.name: tasks(app.tasks) for app in app_configs}
+
+def get_properties(path, app_configs):
+  properties = Properties.load(path)
+  for app_config in app_configs:
+    if app_config.has_portlabel():
+      properties[app_config.name+"_port"] = app_config.get_port()
+  return properties
 
-def render(template, out_file, variables):
+def render(template, out_file, app_configs):
   env = Environment(loader=FileSystemLoader('templates'))
   app_name = template.split(".")[0]
```

Comment:
@lakshmirajagopalan Does it help to rename `app_name` to `template_name`? Since that's essentially what it is.